src/leap/mail/imap/account.py

```diff
@@ -403,34 +449,24 @@ def unsubscribe(self, name):
 
         :param name: name of the mailbox
         :type name: str
+        :rtype: Deferred
         """
-        name = self._parse_mailbox_name(name)
-        if name not in self.subscriptions:
-            raise imap4.MailboxException(
-                "Not currently subscribed to %r" % name)
-        self._set_subscription(name, False)
+        name = normalize_mailbox(name)
 
-    def listMailboxes(self, ref, wildcard):
-        """
-        List the mailboxes.
+        def set_unsubscribed(mbox):
+            return mbox.collection.set_mbox_attr("subscribed", False)
 
-        from rfc 3501:
-        returns a subset of names from the complete set
-        of all names available to the client.  Zero or more untagged LIST
-        replies are returned, containing the name attributes, hierarchy
-        delimiter, and name.
+        d = self.getMailbox(name)
+        d.addCallback(set_unsubscribed)
+        return d
```

Comment:
left TODO. I also realized that unsubscribe should raise an exception if attempted to unsubscribe from non-subscribed mailbox.